data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfiguration.java

```diff
@@ -175,7 +189,19 @@ private void attachSigV4(final RestClientBuilder restClientBuilder) {
     //if not follow regular credentials process
     LOG.info("{} is set, will sign requests using AWSRequestSigningApacheInterceptor", AWS_SIGV4);
     final Aws4Signer aws4Signer = Aws4Signer.create();
-    final AwsCredentialsProvider credentialsProvider = DefaultCredentialsProvider.create();
+    AwsCredentialsProvider credentialsProvider;
+    if (awsStsRole != null && !awsStsRole.isEmpty()) {
+      credentialsProvider = StsAssumeRoleCredentialsProvider.builder()
+              .stsClient(StsClient.create())
+              .refreshRequest(AssumeRoleRequest.builder()
+                      .roleSessionName("Elasticsearch-Sink-" + UUID.randomUUID()
```

Comment:
I mentioned this in the other review.

Should we include the pipeline name here?
e.g.
```
roleSessionName(pipelineName + "-" + UUID.randomUUID().toString())
```

It could be obtained from `pluginSetting.getPipelineName()` in the static factory.

Reply:
+1

Reply:
Missed the comment. Fixed it. in the next push.